Infusionsoft/App.php

```diff
@@ -196,8 +230,16 @@ public function hasTokens(){
     }
 
     public function refreshTokens(){
-        $tokens = Infusionsoft_OAuth2::refreshToken($this->refreshToken);
-        $this->updateAndSaveTokens($tokens['access_token'], $tokens['refresh_token'], $tokens['expires_in']);
+        if (method_exists($this->tokenStorageProvider, 'refreshTokens')){

```

Comment:
Hmm - The alternative that I can see is having a TokenHandler class designation that supersedes using the TokenStorageProvider classes. A TokenHandler class would over-ride all OAuth2 interaction from the SDK (as we are doing with the current implementation of NovakSolutionsInfusionsoftTokenStorageProvider) while TokenStorageProvider would do no more than store and retrieve tokens.  Thoughts?